datadog/fwprovider/data_source_datadog_integration_aws_external_id.go

```diff
@@ -0,0 +1,133 @@
+	d.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	d.Auth = providerData.Auth
+}
+
+func (d *awsIntegrationExternalIDDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "integration_aws_external_id"
+}
+
+func (d *awsIntegrationExternalIDDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve the external ID from an existing AWS integration. This can be used to reference the external ID value from an existing AWS account integration.",
+		Attributes: map[string]schema.Attribute{
+			// Datasource ID
+			"id": utils.ResourceIDAttribute(),
+			"aws_account_id": schema.StringAttribute{
+				Description: "The AWS account ID of the integration to retrieve the external ID from.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[0-9]{12}$`), "invalid aws_account_id"),
+				},
+			},
+			"external_id": schema.StringAttribute{
+				Description: "The external ID associated with the AWS integration.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func (d *awsIntegrationExternalIDDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state awsIntegrationExternalIDDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	awsAccountId := state.AwsAccountId.ValueString()
+
+	// Retrieve AWS account integration for the specific account ID
+	optionalParams := datadogV2.NewListAWSAccountsOptionalParameters().WithAwsAccountId(awsAccountId)
+	accountsResp, httpResp, err := d.Api.ListAWSAccounts(d.Auth, *optionalParams)
+	if err != nil {
+		resp.Diagnostics.Append(utils.FrameworkErrorDiag(utils.TranslateClientError(err, httpResp, "error querying AWS Account Integrations"), ""))
+		return
+	}
+	if err := utils.CheckForUnparsed(accountsResp); err != nil {
+		resp.Diagnostics.AddError("response contains unparsedObject", err.Error())
+		return
+	}
+
+	// Check if we got exactly one account back
+	accounts := accountsResp.GetData()
+	if len(accounts) == 0 {
+		resp.Diagnostics.AddError(
+			"AWS Integration not found",
+			fmt.Sprintf("No AWS integration found for account ID: %s", awsAccountId),
+		)
+		return
+	}
+	if len(accounts) > 1 {
+		resp.Diagnostics.AddError(
+			"Multiple AWS Integrations found",
+			fmt.Sprintf("Unexpected: Multiple integrations found for account ID %s", awsAccountId),
+		)
+		return
+	}
+
+	// Get the single account
+	target := accounts[0]
+	attributes := target.GetAttributes()
+	authConfig, ok := attributes.GetAuthConfigOk()
+	if !ok {
+		resp.Diagnostics.AddError(
+			"No auth config found",
+			fmt.Sprintf("The AWS integration for account %s does not have an auth config", awsAccountId),
+		)
+		return
```

Comment:
⛏️ : this would be a "contact Datadog support" scenario -- this case shouldn't happen under normal circumstances